### PR TITLE
Bugfix/handle exam export multi school permissions

### PR DESCRIPTION
**Pull request by wtritch**
This fixes the naming and grouping services used during organization exam exporting to handle the case when a user has permissions for multiple schools but not for the schools' school-groups or districts.

### CODE REVIEW

File: webapp/src/main/webapp/src/app/organization-export/organization-export-naming.service.ts [hunk 1]
```diff
@@ -52,10 +52,14 @@ export class OrganizationExportNamingService {
...
       if (!Utils.isUndefined(selection.firstSchool.schoolGroupId)
+        && !Utils.isUndefined(selection.firstSchool.schoolGroup)
         && selection.schools.every(x => x.schoolGroupId === selection.firstSchool.schoolGroupId)) {
         return selection.firstSchool.schoolGroup.name;
       }
-      return selection.firstSchool.district.name;
+      if (!Utils.isUndefined(selection.firstSchool.district)) {
+        return selection.firstSchool.district.name;
+      }
+      return selection.firstSchool.name;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/organization-export/organization-export-naming.service.ts]: revisiting this i think we should really just make the user name the thing. coding these smart names is pretty hairy.

File: webapp/src/main/webapp/src/app/organization-export/organization-grouping.service.ts [hunk 2]
```diff
@@ -85,11 +86,23 @@ export class OrganizationGroupingService {
...
+  /**
+   * Test if this is a valid grouping node.  While building the organization tree we create shallow
+   * placeholders for districts and school groups that the user does not have explicit access to.  These
+   * shallow nodes should not be used for grouping.
+   *
+   * @param {Tree<Organization>} node An organizaiton node
+   * @returns {boolean} True if the node is a valid grouping node
+   */
+  private validGroupingNode(node: Tree<Organization>) {
+    return !isNullOrUndefined(node.value.name);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/organization-export/organization-grouping.service.ts]: is this null-safe?
I would expect:
```typescript
return !isNullOrUndefined(node.value) && !isNullOrUndefined(node.value.name);
```

File: webapp/src/main/webapp/src/app/organization-export/organization-export-naming.service.ts (hunk 1 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/organization-export/organization-export-naming.service.ts]: +1 but not today.  Heuristics always end poorly IMHO =)

File: webapp/src/main/webapp/src/app/organization-export/organization-grouping.service.ts (hunk 2 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/organization-export/organization-grouping.service.ts]: yes, the value always exists (even if it's shallow and just has an "id" property)

Comment (esotrope) [reply webapp/src/main/webapp/src/app/organization-export/organization-grouping.service.ts]: ok cools - thanks

File: webapp/src/main/webapp/src/app/organization-export/organization-export-naming.service.ts (hunk 1 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/organization-export/organization-export-naming.service.ts]: ill make a jira for it

**Pull request merged**